Several task tests failed in CI

Looks like RunAsyncAdditionalBehaviorsTests_NegativeCases2 and one of the TaskCancelWaitTest tests.

http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_debug_prtest/3082/console
```
Discovering: System.Threading.Tasks.Tests
  Discovered:  System.Threading.Tasks.Tests
  Starting:    System.Threading.Tasks.Tests
  
  Unhandled Exception: Xunit.Sdk.TrueException:      > OCE shouldn't have resulted in unobserved event firing with System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> System.InvalidOperationException: Invalid attempt made to decrement the event's count below zero.
     at System.Threading.CountdownEvent.Signal(Int32 signalCount)
     at System.Threading.Tasks.Tests.CancelWait.TaskCancelWaitTest.<>c__DisplayClass1.<CreateTask>b__0(Object o) in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Threading.Tasks\tests\Task\TaskCancelWaitTest.cs:line 167
     at System.Threading.Tasks.Task.InnerInvoke()
     at System.Threading.Tasks.Task.Execute()
     --- End of inner exception stack trace ---
  ---> (Inner Exception #0) System.InvalidOperationException: Invalid attempt made to decrement the event's count below zero.
     at System.Threading.CountdownEvent.Signal(Int32 signalCount)
     at System.Threading.Tasks.Tests.CancelWait.TaskCancelWaitTest.<>c__DisplayClass1.<CreateTask>b__0(Object o) in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Threading.Tasks\tests\Task\TaskCancelWaitTest.cs:line 167
     at System.Threading.Tasks.Task.InnerInvoke()
     at System.Threading.Tasks.Task.Execute()<---
  
  Expected: True
  Actual:   False
     at Xunit.Assert.True(Nullable`1 condition, String userMessage)
     at Xunit.Assert.True(Boolean condition, String userMessage)
     at System.Threading.Tasks.Tests.AsyncTaskMethodBuilderTests.<RunAsyncAdditionalBehaviorsTests_NegativeCases2>b__3d(Object s, UnobservedTaskExceptionEventArgs e)
     at System.Threading.Tasks.TaskScheduler.PublishUnobservedTaskException(Object sender, UnobservedTaskExceptionEventArgs ueea)
     at System.Threading.Tasks.TaskExceptionHolder.Finalize()
d:\j\workspace\dotnet_corefx_windows_debug_prtest\packages\Microsoft.DotNet.BuildTools.1.0.25-prerelease-00070\lib\tests.targets(130,5): warning MSB3073: The command "CoreRun.exe xunit.console.netcore.exe System.Threading.Tasks.Tests.dll  -xml testResults.xml -notrait category=nonwindowstests  -notrait category=OuterLoop -notrait category=failing " exited with code -532462766. [d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Threading.Tasks\tests\System.Threading.Tasks.Tests.csproj]
d:\j\workspace\dotnet_corefx_windows_debug_prtest\packages\Microsoft.DotNet.BuildTools.1.0.25-prerelease-00070\lib\tests.targets(141,5): error : One or more tests failed while running tests from 'System.Threading.Tasks.Tests' please check log for details! [d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Threading.Tasks\tests\System.Threading.Tasks.Tests.csproj]
  System.Threading.Timer.Tests -> d:\j\workspace\dotnet_corefx_windows_debug_prtest\bin\Windows_NT.AnyCPU.Debug\System.Threading.Timer.Tests\System.Threading.Timer.Tests.dll
```

--- a/src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/AsyncTaskMethodBuilderTests.cs
+++ b/src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/AsyncTaskMethodBuilderTests.cs
@@ -434,22 +434,16 @@ namespace System.Threading.Tasks.Tests
                 cts.Cancel();
                 b.SignalAndWait(); // release task to complete
 
-                // A TCS task
-                var tcs = new TaskCompletionSource<int>();
-                tcs.SetCanceled();
-                Task t2 = tcs.Task;
-
-                EventHandler<UnobservedTaskExceptionEventArgs> handler = (s, e) =>
-                {
-                    Assert.True(false, string.Format("     > OCE shouldn't have resulted in unobserved event firing with " + e.Exception.ToString()));
-                };
+                // This test may be run concurrently with other tests in the suite, 
+                // which can be problematic as TaskScheduler.UnobservedTaskException
+                // is global state.  The handler is carefully written to be non-problematic
+                // if it happens to be set during the execution of another test that has 
+                // an unobserved exception.
+                EventHandler<UnobservedTaskExceptionEventArgs> handler = 
+                    (s, e) => Assert.DoesNotContain(oce, e.Exception.InnerExceptions);
                 TaskScheduler.UnobservedTaskException += handler;
-
                 ((IAsyncResult)t1).AsyncWaitHandle.WaitOne();
-                ((IAsyncResult)t2).AsyncWaitHandle.WaitOne();
                 t1 = null;
-                t2 = null;
-
                 for (int i = 0; i < 10; i++)
                 {
                     GC.Collect();

